accord-core/src/main/java/accord/local/CommandStore.java

```diff
@@ -83,6 +123,69 @@ public Agent agent()
 
     public abstract void shutdown();
 
+    // implementations are expected to override this for persistence
+    protected void setRejectBefore(ReducingRangeMap<Timestamp> newRejectBefore)
+    {
+        this.rejectBefore = newRejectBefore;
+    }
+
+    /**
+     * To be overridden by implementations, to ensure the new state is persisted
+     *
+     * TODO (required): consider handling asynchronicity of persistence
+     *  (could leave to impls to call this parent method once persisted)
+     * TODO (desired): compact Ranges, merging overlaps
+     */
+    protected void setBootstrapBeganAt(NavigableMap<TxnId, Ranges> newBootstrapBeganAt)
+    {
+        this.bootstrapBeganAt = newBootstrapBeganAt;
+    }
+
+    protected abstract void registerHistoricalTransactions(Deps deps);
+<<<<<<< Updated upstream
+=======
+    protected abstract void evictLog(TxnId txnIdBefore);
+>>>>>>> Stashed changes
```

Comment:
unresolved conflict

Reply:
Oh, whoops that seems to have peeked into the future